src/ParseNatSpec.js

```diff
@@ -0,0 +1,75 @@
+
+/**
+ * Parse a full text comment to an object that can be easly consumed
+ * @param comment full text comment
+ */
+module.exports = function (comment) {
+  let isMultiline = false;
+
+  // remove comments definers
+  // the comments can be multiline of single line
+  if (comment.substring(0, 3) === '/**') {
+    isMultiline = true;
+    // if it is a multiline, remove line breaks, "/**"" (the comment begin)
+    // "*/"" (the comment end), and "* @" at the begining of each comment
+    comment = comment.replace(/(\r|\n|\/\*\*|\*\/|\*(?=[ ]?@))/g, '');
+  } else if (comment.substring(0, 3) === '///') {
+    // if it is a single line,
+    // remove just line breakers and "///" (the comment begin)
+    comment = comment.replace(/(\r|\n|\/\/\/)/g, '');
+  }
+
+  // split text by comment type
+  const splitComments = comment
+    .split(/@(title|author|notice|dev|param|return)/g);
+  // let's start a map
+  const resultComments = new Map();
```

Comment:
Why not use a simple object literal here?

Reply:
right, much better. Thank you